src/index.ts

```diff
@@ -101,12 +101,12 @@ function convertBundleIntoReact(
   }
 
   const content: React.ReactNode[] = [];
-  const linkRegex = /(\s+|^)(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})(\s+|$)/g;
+  const linkRegex = /(\s+|^)(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/g;
 
   let index = 0;
   let match: RegExpExecArray | null;
   while ((match = linkRegex.exec(bundle.content)) !== null) {
-    const [ , pre, url, post ] = match;
+    const [ , pre, url ] = match;
```

Comment:
Not strictly related to this PR, but how about using named capturing groups (`(?<Name>x)`) in the regex?

Reply:
Safia suggested this in the original PR, but since support for named capture groups is still below 90% (albeit barely) I decided to use this array unpacking for readability instead.